src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStationProxy.cs

```diff
@@ -0,0 +1,273 @@
+    {
+        private static readonly Dictionary<SynologyApi, string> Resources;
+
+        private readonly IHttpClient _httpClient;
+        private readonly Logger _logger;        
+        private bool _authenticated;
+
+        static DownloadStationProxy()
+        {
+            Resources = new Dictionary<SynologyApi, string>
+            {
+                {SynologyApi.Info, "query.cgi" },
+            };
+        }
+
+        public DownloadStationProxy(ICacheManager cacheManager, IHttpClient httpClient, Logger logger)
+        {            
+            this._httpClient = httpClient;
+            this._logger = logger;
+        }
+
+        public void AddTorrentFromData(byte[] torrentData, string downloadDirectory, DownloadStationSettings settings)
+        {
+            var arguments = new Dictionary<string, object>
+            {
+                { "api", "SYNO.DownloadStation.Task"},
+                { "version", "2"},
+                { "method", "create"},
+                { "file", Convert.ToBase64String(torrentData)}
+            };
+
+            if (downloadDirectory.IsNotNullOrWhiteSpace())
+            {
+                arguments.Add("destination", downloadDirectory);
+            }
+
+            ProcessRequest(SynologyApi.DownloadStationTask, arguments, settings, HttpMethod.POST);
+        }
+
+        public void AddTorrentFromUrl(string torrentUrl, string downloadDirectory, DownloadStationSettings settings)
+        {
+            var arguments = new Dictionary<string, object>
+            {
+                { "api", "SYNO.DownloadStation.Task"},
+                { "version", "3"},
+                { "method", "create"},
+                { "uri", torrentUrl}
+            };
+
+            if (downloadDirectory.IsNotNullOrWhiteSpace())
+            {
+                arguments.Add("destination", downloadDirectory);
+            }
+
+           var b= ProcessRequest(SynologyApi.DownloadStationTask, arguments, settings, HttpMethod.GET);
+        }
+
+        public IEnumerable<DownloadStationTask> GetItems(DownloadStationSettings settings)
+        {
+            var arguments = new Dictionary<string, object>
+            {
+                 { "api", "SYNO.DownloadStation.Task"},
+                 { "version", "1"},
+                 { "method", "list"},
+                 { "additional", "detail,transfer"}
+            };
+
+            var result = ProcessRequest<DownloadStationTaskInfoResponse>(SynologyApi.DownloadStationTask, arguments, settings);
+
+            if (result.Success)
+            {
+                return result.Data.Tasks.Where(t => t.Type == DownloadStationTask.DownloadStationTaskType.BT);
+            }
+
+            return new List<DownloadStationTask>();
+        }
+
+        public Dictionary<string, object> GetConfig(DownloadStationSettings settings)
+        {
+            var arguments = new Dictionary<string, object>
+            {
+                {"api", "SYNO.DownloadStation.Info"},
```

Comment:
`{ "api", "SYNO.DownloadStation.Info" },` (and the same for the following lines).